### PR TITLE
Remove Keychain.currentSession

### PR DESCRIPTION
**Pull request by vanwagonet**
environment.currentSession is the source of truth for logged in state.
Don't have multiple sources of truth, so we don't have to keep them in sync.

extra: align window setup & keychain management

refs: none
affects: none
release note: none

### CODE REVIEW

File: Parent/Parent/ParentAppDelegate.swift [hunk 1]
```diff
@@ -50,13 +46,17 @@ class ParentAppDelegate: UIResponder, UIApplicationDelegate {
         return vc
     }
 
+    lazy var environment: AppEnvironment = {
+        let env = AppEnvironment.shared
+        env.router = Parent.router
+        // env.backgroundAPIManager.eventsHandler = self
```
Comment (armstrongnate) [inline Parent/Parent/ParentAppDelegate.swift]: `backgroundAPIManager` is no longer a thing. You can remove this comment.

File: Canvas/Canvas/CanvasAppDelegate.swift [hunk 2]
```diff
@@ -29,9 +29,9 @@ import Core
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate, AppEnvironmentDelegate {
+    var window: UIWindow? = MasqueradableWindow(frame: UIScreen.main.bounds)
```
Comment (vanwagonet) [inline Canvas/Canvas/CanvasAppDelegate.swift]: TIL it needs to be optional to conform to `UIApplicationDelegate.window`

File: Canvas/Canvas/CanvasAppDelegate.swift [hunk 3]
```diff
@@ -348,11 +347,11 @@ extension AppDelegate: NativeLoginManagerDelegate {
     }
     
     func didLogout(_ controller: UIViewController) {
-        if let entry = Keychain.currentSession {
-            Keychain.removeEntry(entry)
+        if let entry = environment.currentSession {
             environment.userDidLogout(session: entry)
             CoreWebView.stopCookieKeepAlive()
         }
+        Keychain.clearEntries()
```
Comment (vanwagonet) [inline Canvas/Canvas/CanvasAppDelegate.swift]: until an app is actually logging in with the new login process, every logout clears all keychain entries.

File: Parent/Parent/ParentAppDelegate.swift [hunk 4]
```diff
@@ -29,12 +29,8 @@ let ParentAppRefresherTTL: TimeInterval = 5.minutes
 
 @UIApplicationMain
 class ParentAppDelegate: UIResponder, UIApplicationDelegate {
-
-    var environment: AppEnvironment!
-
-    var window: UIWindow?
+    var window: UIWindow? = MasqueradableWindow(frame: UIScreen.main.bounds)
     var legacySession: Session?
-    var session: KeychainEntry?
```
Comment (vanwagonet) [inline Parent/Parent/ParentAppDelegate.swift]: this was another potentially conflicting current session.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 94.18% | 94.19% | -0.02%
**Student** | 94.07% | 93.58% | 0.48%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.54% | 92.54% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**